packer/install-dependencies.sh

```diff
@@ -182,15 +182,12 @@ for KUBE_DNS_MASQ_VERSION in ${KUBE_DNS_MASQ_VERSIONS}; do
     echo "  - ${CONTAINER_IMAGE}" >> ${VHD_LOGS_FILEPATH}
 done
 
-PAUSE_VERSIONS="3.1"
+PAUSE_VERSIONS="1.2.0"
```

Comment:
How confident are we that this new pause image is backwards compatible with all previous versions of k8s?

Reply:
uhm... I had thought about this a bit and reasoned that we only had one version difference used and that was in: https://github.com/Azure/aks-engine/pull/2050/files/096ad5d32ac7ba931f36876d1b7131a2823ae1aa#diff-bb9b6cb6f08a63800be959ba49eaf714L364. Everything else was pause:3.1. 

Based on the only difference being 3.0 and 3.1, I figured pause was following semantic versioning and thus would not have a breaking change between 3.0 and 3.1. Granted, could be more credit than due, but seemed reasonable.

Based on the above, I figured there is relatively low risk of using only pause:1.2.0.

What do you think?

Reply:
Are we saying that this particular `1.2.0` release has an equivalent "3 dot" release (currently we are using `3.1` from `k8s.gcr.io`.

In addition, the k8s.gcr.io pause image is the one referenced inside core components, so if we remove that image from being pre-pulled we break disconnected scenarios.

Reply:
I have added to be pre-pulled in `install_dependencies.sh`.

AFAICT, the image works with the existing tests. I believe, @jchauncey is building this image to be a replacement for the `k8s.gcr.io` image.

Reply:
What I'm saying is: are we confident that all the changes in this PR mean that all cluster configurations will no longer refer to the k8s.gcr.io pause 3.1 image?

Reply:
I'm not confident that all cluster configs will only use mcr pause.

Since that is the case, should we then continue to pull `k8s.gcr.io`? Should I run each config without outbound network access? What would you recommend?

Reply:
I recommend we add the new pause image to our VHD config spec, and keep the existing pause image as well.

(Generally we should be keeping existing pre-pulled images to protect the disconnected environments for back-compat scenarios; hyperkube images are an exception because we don't have the storage overhead to keep older than n-1 images around.)

Reply:
Ok, so both are now getting pulled in `install_dependencies.sh`.

https://github.com/Azure/aks-engine/pull/2050/files#diff-62ed5765f6aaf1d428746b3d7b597100R191-R204